lib/hostip.c

```diff
@@ -652,6 +667,35 @@ static bool tailmatch(const char *full, const char *part)
   return strncasecompare(part, &full[flen - plen], plen);
 }
 
+static bool is_hostname_blocked(struct Curl_easy *data, const char *hostname)
```

Comment:
This doesn't scale very well. In many situations (such as blocking spam) you could have hundred of thousands of entries in the blocklist. I'd much rather see a solution using a bloom filter or similar.

Reply:
I thought about that, too. However, I think implementing something more complex might drive this pull request out-of-scope for its intended purpose: provide the plumbing for making dynamic the blocklisting of domains.

I suspect the underlying data structures and algorithms could be improved upon at a later stage of development. I also think this tiny little feature won't be used much, or with large numbers of entries, in its early days.